src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js

```diff
@@ -182,6 +171,28 @@ export class VariableService {
         return variablesWithContextAndConstantVariables;
     }
 
+    updateVariable(variableToAssign: string, data: any, variablesHash: RuleVariables) {
+        const variableHashKey = variableToAssign.replace('#{', '').replace('A{', '').replace(/}/g, '');
+        const variableHash = variablesHash[variableHashKey];
+
+        if (!variableHash) {
+            // If a variable is mentioned in the content of the rule, but does not exist in the variables hash, show a warning:
+            log.warn(`Variable ${variableHashKey} was not defined.`);
+        } else {
+            const { variableType } = variableHash;
+            const variableValue = data === null ? this.defaultValues[variableType] : normalizeRuleVariable(data, variableType);
+
+            variablesHash[variableHashKey] = {
+                ...variableHash,
+                variableValue,
+                hasValue: data !== null,
```

Comment:
The `IConvertInputRulesValue` type definition also allows `undefined` to be the output from the converter. I don't think we are converting undefined to null anywhere (but this might be the solution). Can you look into what happens if `undefined` is returned from the converter and potentially how we can best deal with it?